### PR TITLE
Fixes #762: Make API to bookmark skills

### LINKED ISSUE #762: Make API to bookmark skills
**Actual Behaviour**
No such API present.

**Expected Behaviour**
Make an API to favourite skills.

**Steps to reproduce it**
NA

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by akshatnitd**
Fixes #762 

Changes: 
Make API to bookmark skills

Screenshots for the change: 
* bookmarkSkill.json
```
{
	"abc@a.com": {
		"general": {
			"Knowledge": {
				"en": {
					"aboutsusi": {
						"bookmark": true,
						"timestamp": "2018-06-23 15:50:00.3"
					}
				}
			}
		}
	},
	"ab@a.com": {
		"general": {
			"Knowledge": {
				"en": {
					"aboutsusi": {
						"bookmark": true,
						"timestamp": "2018-06-23 15:53:41.632"
					},
					"acronyms": {
						"bookmark": true,
						"timestamp": "2018-06-23 15:53:12.334"
					}
				}
			}
		}
	}
}```

### CODE REVIEW

File: src/ai/susi/server/api/cms/BookmarkSkillService.java [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+ * bookmark can be 0 or 1
+ * http://localhost:4000/cms/bookmarkSkill.json?model=general&group=Knowledge&skill=aboutsusi&bookmark=1&access_token=6O7cqoMbzlClxPwg1is31Tz5pjVwo3
+ */
+public class BookmarkSkillService extends AbstractAPIHandler implements APIHandler {
+
+
+	private static final long serialVersionUID = -1960932190918215684L;
+
+
+	@Override
```
Comment (Akshat-Jain) [inline src/ai/susi/server/api/cms/BookmarkSkillService.java]: Please correct the indentation here.

File: src/ai/susi/server/api/cms/BookmarkSkillService.java [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+/**
+ * This Endpoint accepts 5 parameters. model,group,language,skill,bookmark.
+ * bookmark can be 0 or 1
+ * http://localhost:4000/cms/bookmarkSkill.json?model=general&group=Knowledge&skill=aboutsusi&bookmark=1&access_token=6O7cqoMbzlClxPwg1is31Tz5pjVwo3
+ */
+public class BookmarkSkillService extends AbstractAPIHandler implements APIHandler {
+
+
+	private static final long serialVersionUID = -1960932190918215684L;
```
Comment (Akshat-Jain) [inline src/ai/susi/server/api/cms/BookmarkSkillService.java]: Please correct the indentation here.

File: src/ai/susi/server/api/cms/BookmarkSkillService.java [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+import java.sql.Timestamp;
+
+
+/**
+ * This Endpoint accepts 5 parameters. model,group,language,skill,bookmark.
+ * bookmark can be 0 or 1
+ * http://localhost:4000/cms/bookmarkSkill.json?model=general&group=Knowledge&skill=aboutsusi&bookmark=1&access_token=6O7cqoMbzlClxPwg1is31Tz5pjVwo3
+ */
+public class BookmarkSkillService extends AbstractAPIHandler implements APIHandler {
+
```
Comment (Akshat-Jain) [inline src/ai/susi/server/api/cms/BookmarkSkillService.java]: Remove unnecessary blank lines at all places.

Comment (akshatnitd) [reply src/ai/susi/server/api/cms/BookmarkSkillService.java]: Done!

File: src/ai/susi/server/api/cms/BookmarkSkillService.java (hunk 1 above)
Comment (akshatnitd) [reply src/ai/susi/server/api/cms/BookmarkSkillService.java]: Fixed all indentation issues

**Comment by akshatnitd** [comment]
Conflicts fixed. Will also be sending the changes suggested by @AnupKumarPanwar today.

**Comment by akshatnitd** [comment]
@AnupKumarPanwar Please have a look once. Have updated the PR description with the updated JSON structure.

**Comment by AnupKumarPanwar** [comment]
I suggest we should use JSON Array in the structure. Something like this.
```
{
	"user1_id" : [
		{bookmarked_skill_1},
		{bookmarked_skill_2},
	],
	"user2_id" : [
		{bookmarked_skill_1},
		{bookmarked_skill_2},
	]
}
```

**Comment by Orbiter** [comment]
@AnupKumarPanwar if we should do the change as you suggested in https://github.com/fossasia/susi_server/pull/829#issuecomment-399664496 please open a follow-up issue

**Pull request merged**